fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/mixin/event/interaction/MixinServerPlayerInteractionManager.java

```diff
@@ -81,4 +87,28 @@ public void interactItem(ServerPlayerEntity player, World world, ItemStack stack
 			return;
 		}
 	}
+
+	@Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/block/Block;onBreak(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/BlockState;Lnet/minecraft/entity/player/PlayerEntity;)V"), method = "tryBreakBlock", locals = LocalCapture.CAPTURE_FAILHARD, cancellable = true)
+	private void breakBlock(BlockPos pos, CallbackInfoReturnable<Boolean> cir, BlockState state, BlockEntity entity, Block block) {
+		ActionResult result = PostBreakBlockCallback.EVENT.invoker().beforeBlockBreak(pos, state, entity, block);
+
+		if (result == ActionResult.FAIL) {
+			BlockPos cornerPos = pos.add(-1, -1, -1);
+
+			for (int x = 0; x < 3; x++) {
+				for (int y = 0; y < 3; y++) {
+					for (int z = 0; z < 3; z++) {
+						this.player.networkHandler.sendPacket(new BlockUpdateS2CPacket(world, cornerPos.add(x, y, z)));
```

Comment:
Are we really to send 27 packets? Like is 7 packets not sufficient (send packet at pos and do a for loop for direction values and send packet for pos offset at each direction)

Also if you do need to send by radius, there is method in block pos that allow you to iterate block pos in a radius (3d) from closest to furthest.

Reply:
I tested that, and yes u do need all 27 packets because of redstone

Reply:
Does redstone dust update block that far? And how can we make sure modded blocks don't update further?

Reply:
does vanilla do this?

Reply:
Does redstone dust update block that far? And how can we make sure modded blocks don't update further?
we don't, mods will just have to send packets most likely, but it might not be an issue I'm not sure.

and redstone dust

X
BX

x = redstone dust
b = block
break the bottom redstone dust, the top one will change it's state, that's why we need the 27.
if your concern is performance this is nothing I wouldn't worry about it

Reply:
but isnt this the failed condition? failed shouldn't change anything right

Reply:
I still don't understand why the redstone will get updated if the block didn't change